### DISCUSSION TITLE
How do i get the filename in /api/anyApi file ??

### DISCUSSION THREAD

**Discussion by sahilrajput03** [vercel/next.js#23692] [Help]
`console.log(__filename) // Output: /index.js`

What is the work i need to do here..?

**Comment by akellbl4** [comment]
Hi, what's wrong with `console.log(__filename)`?
Next.js API routes are simple nodejs functions and you can use any nodejs functions and variables in it. I just tried to get the name of a file and `__filename` works just fine ;)

**Reply by sahilrajput03** [reply]
> Hi, what's wrong with `console.log(__filename)`?
> Next.js API routes are simple nodejs functions and you can use any nodejs functions and variables in it. I just tried to get the name of a file and `__filename` works just fine ;)

I mean to do it in /api directory files..,?
Did you try it in there ?

**Reply by sahilrajput03** [reply]
![image](https://user-images.githubusercontent.com/31458531/114926178-a9995200-9e4d-11eb-8b3b-985f1a98c78c.png)

Check the codesandbox [here](https://codesandbox.io/s/build-straightforward-api-with-nextjs-api-routes-forked-nextjs-route-filename-error-hbp3k).

**Reply by sahilrajput03** [reply]
@akellbl4 ? I don't think __filename works at all.

**Reply by sahilrajput03** [reply]
Although, the way i managed to figure out the filename is via 
```
const hello: NextApiHandler = (req, res) => {
  const file_name = req.url;
```

**Reply by akellbl4** [reply]
`__filename` contains just the name of the file.
If you need to have pathname from URL `req.url` works well.

**Reply by zargold** [reply]
Not sure when this was updated but I would say _filename provides a lot more information than just the actual filename:
``` js
// [slug].ts
export const getServerSideProps = async ({ req }) => {

console.log({ __filename, 'req.url': req.url });
const resultOfLogging = {
  __filename: '/my-next-js-project/.next/server/pages/blog/[slug].js',
  'req.url': '/blog/bargerg',
}
 return {
 };
}
```
Note I use typescript.

**Comment by 1mlm** [comment]
I know this comment section is old, but for me. I use Next.JS + Vercel. With this structure on (app/api/):
api/
  - db/
  - - addition.ts
  - - substraction.ts
  - defs.ts
 
When calling the API from the client side (app/exercises/page.ts), I get this. Litterally can't get the name bc of webpack's dynamic naming hasing or something i suppose
```ts
console.log(__filename) // C:\Users\ptdrr\Desktop\Projets\bimowy\.next\server\chunks\[root of the server]__47d6b187._.js
console.log(path.basename(__filename)) // [root of the server]__47d6b187._.js
console.log(path.parse(__filename))
// {
//   root: 'C:\\',
//   dir: 'C:\\Users\\ptdrr\\Desktop\\Projets\\bimowy\\.next\\server\\chunks',
//   base: '[root of the server]__47d6b187._.js',
//   ext: '.js',
//   name: '[root of the server]__47d6b187._'
// }
```
